sdk/core/System.BinaryData/src/BinaryData.cs

```diff
@@ -0,0 +1,260 @@
+            bool async,
+            CancellationToken cancellationToken = default)
+        {
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+            if (stream.CanSeek && stream.Length > int.MaxValue)
+            {
+                throw new ArgumentOutOfRangeException(
+                    nameof(stream),
+                    "Stream length must be less than Int32.MaxValue");
+            }
+            using var memoryStream = new MemoryStream();
+            if (async)
+            {
+                await stream.CopyToAsync(memoryStream, CopyToBufferSize, cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                stream.CopyTo(memoryStream);
+            }
+            return new BinaryData(memoryStream.GetBuffer().AsMemory(0, (int) memoryStream.Position));
+        }
+
+
+        /// <summary>
+        /// Creates a <see cref="BinaryData"/> instance by serializing the passed in object using
+        /// the <see cref="JsonSerializer"/>.
+        /// </summary>
+        ///
+        /// <typeparam name="T">The type of the data.</typeparam>
+        /// <param name="jsonSerializable">The data to use.</param>
+        /// <param name="options">The <see cref="JsonSerializerOptions"/> to use when serializing to Json.</param>
+        ///
+        /// <returns>A <see cref="BinaryData"/> instance.</returns>
+        [Diagnostics.CodeAnalysis.SuppressMessage("Usage", "AZC0014:Avoid using banned types in public API", Justification = "<Pending>")]
+        public static BinaryData FromObjectAsJson<T>(
+            T jsonSerializable,
+            JsonSerializerOptions? options = default)
+        {
+            using var memoryStream = new MemoryStream();
+            var buffer = JsonSerializer.SerializeToUtf8Bytes(jsonSerializable, typeof(T), options);
+            memoryStream.Write(buffer, 0, buffer.Length);
+            return new BinaryData(memoryStream.GetBuffer().AsMemory(0, (int) memoryStream.Position));
+        }
+
+        /// <summary>
+        /// Converts the <see cref="BinaryData"/> to a string using UTF-8.
+        /// </summary>
+        /// <returns>The string representation of the <see cref="BinaryData"/> using UTF-8
+        /// to decode the bytes.</returns>
+        public override string ToString()
+        {
+            if (MemoryMarshal.TryGetArray(
+                Bytes,
+                out ArraySegment<byte> data))
+            {
+                return s_encoding.GetString(data.Array, data.Offset, data.Count);
+            }
+            return s_encoding.GetString(Bytes.ToArray());
+        }
+
+        /// <summary>
+        /// Converts the <see cref="BinaryData"/> to a read-only stream.
+        /// </summary>
+        /// <returns>A stream representing the data.</returns>
+        public Stream ToStream() =>
+            new ReadOnlyMemoryStream(Bytes);
```

Comment:
Why create a whole new type for it? I'd just expect it to be `new MemoryStream(_bytes)`.

Reply:
We didn't want customers depending on it being a MemoryStream.